publish/dir.props

```diff
@@ -5,6 +5,9 @@
   <PropertyGroup>
     <ContainerName Condition="'$(ContainerName)' == ''">dotnet</ContainerName>
     <NuGetPushTimeoutSeconds Condition="'$(NuGetPushTimeoutSeconds)' == ''">3600</NuGetPushTimeoutSeconds>
+    <BinariesRelativePath>$(Channel)/Binaries/$(SharedFrameworkNugetVersion)</BinariesRelativePath>
```

Comment:
This should all be `$(ProductVersion)`.

I'm still not sure why we have both, but when we are talking generally about the build number/version, use `$(ProductVersion)`.  When you are specifically talking about the SharedFramework or MS.NC.App, use `$(SharedFrameworkNuGetVersion)`